templates/firstboot/mellanox_fw_update.yaml

```diff
@@ -230,7 +263,30 @@ resources:
                                       " please update to the latest mstflint package." % key)
                             continue
 
-                        if current_mlx_config.get(key) and value.lower(
+                        if key in self.arrayed_mlnx_config.keys():
+                            if value.get('*'):
+                                # Aggregate required configuration for all the indexes
+                                first_index = int(self.arrayed_mlnx_config[key][0])
+                                last_index = int(self.arrayed_mlnx_config[key][1]) + 1
+                                param_value = value.get('*')
+                                for i in range(first_index, last_index):
+                                    param_key = "%s[%s]" % (key, i)
+                                    if current_mlx_config.get(param_key) and \
+                                            param_value.lower() not in \
+                                            current_mlx_config.get(param_key).lower():
+                                        params_to_set.append("%s=%s" % (param_key,
+                                                                        param_value))
+                            else:
+                                 # Aggregate required configuration to specific indexes
+                                 for key_index, param_value in value.items():
+                                     param_key = "%s[%s]" % (key, key_index)
+                                     if current_mlx_config.get(param_key) and \
+                                             str(param_value).lower() not in \
```

Comment:
is there a reason you are using `in` and not `==` ?
e.g  "1" in "100" is true and would then lead to the configuration to be skipped